Test/NumberTheory/fastPower.test.cpp

```diff
@@ -0,0 +1,24 @@
+#define FAST_POWER_TEST
+#define CATCH_CONFIG_MAIN
+
+#include "../catch.hpp"
+#include "../../NumberTheory/FastPower.cpp"
+
+
+TEST_CASE("Base cases", "[fast_power]") {
+    REQUIRE(fast_power(2,2) == 4);
+    REQUIRE(fast_power(2,4) == 16);
+}
+
+TEST_CASE("Normal cases", "[fast_power]") {
+    REQUIRE(fast_power(3,4) == 81);
+    REQUIRE(fast_power(7,9) == 40353607);
+    REQUIRE(fast_power(15,10) == 576650390625);
+}
+
+TEST_CASE("Overflow cases", "[fast_power]") {
+    REQUIRE(fast_power(2,100) == 0);
+    REQUIRE(fast_power(10,99) == 0);
```

Comment:
I think you meant ``` fast_power(2 ,100) ``` in the first one.
Anyway, regarding large numbers, to fix the output being ```0``` we can output them modulo a large number such as ```1000000000007``` , but not a huge number like ```2^64```. Where ```fast_power(2, 100)``` will be ```40955801449``` .
Regarding the last statement, do you mean changing ```ull``` to ```Ull``` ?